### PR TITLE
python311Packages.objax: disable tests to fix build

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

I cannot get the tests do run (duplicate packages in the closure)... so I disabled them.

In case anyone has an idea, here are the logs when I tried to add `pytestCheckHook` and `tensorflow` to `nativeCheckInputs`:

<details>

cc @ndl 

```
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests
no Makefile or custom installCheckPhase, doing nothing
pythonCatchConflictsPhase
/nix/store/q2zpnf017gccs9w9mb0ikdqhkf0k5v82-catch_conflicts.py:1: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  import pkg_resources
Found duplicated packages in closure for dependency 'grpcio':
  grpcio 1.27.3 (/nix/store/kvznrm3za4rxf3q106jq3q018rh65dyr-python3.11-grpcio-1.27.3/lib/python3.11/site-packages)
  grpcio 1.57.0 (/nix/store/0n366229mzg5ncdpk980fg07s89bsxvf-python3.11-grpcio-1.57.0/lib/python3.11/site-packages)
Found duplicated packages in closure for dependency 'protobuf':
  protobuf 4.21.12 (/nix/store/hngkl1d0n2j0cw3nv32jadgrfmqwy0c3-python3.11-protobuf-4.21.12/lib/python3.11/site-packages)
  protobuf 4.24.3 (/nix/store/6yd51vrzfmnzkr6pn7f8kwarwgdqwf8q-python3.11-protobuf-4.24.3/lib/python3.11/site-packages)
Found duplicated packages in closure for dependency 'tensorboard':
  tensorboard 2.14.0 (/nix/store/7kif9z36wy7z1hmnhkfzkp53l53aidcz-python3.11-tensorboard-2.14.0/lib/python3.11/site-packages)
  tensorboard 2.14.0 (/nix/store/5pm36bkbwgrrivhwd81kw493kkdippvz-python3.11-tensorboard-2.14.0/lib/python3.11/site-packages)
Found duplicated packages in closure for dependency 'tensorboard-plugin-profile':
  tensorboard-plugin-profile 2.11.1 (/nix/store/10d5h4x52znxm77n1aghm8bbaj8pzagw-python3.11-tensorboard_plugin_profile-2.11.1/lib/python3.11/site-packages)
  tensorboard-plugin-profile 2.11.1 (/nix/store/39g69ya6495jv0iszkww5ppjqnhnqp4x-python3.11-tensorboard_plugin_profile-2.11.1/lib/python3.11/site-packages)

Package duplicates found in closure, see above. Usually this happens if two packages depend on different version of the same dependency.
```

</details>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/objax/default.nix [hunk 1]
```diff
@@ -40,6 +40,10 @@ buildPythonPackage rec {
     "objax"
   ];
 
+  # Tests require tensorflow which causes conflicts in the closure due to duplicates of the
+  # following packages: grpcio, protobuf, tensorboard and tensorboard-plugin-profile.
+  doCheck = false;
```
Comment (SomeoneSerge) [inline pkgs/development/python-modules/objax/default.nix:45]: Have you seen the approach in https://github.com/NixOS/nixpkgs/blob/3fdb70efbba3047c2ed2cceacf3d63ae4e5bdf5f/pkgs/development/python-modules/cleanlab/default.nix#L47-L52 (https://github.com/NixOS/nixpkgs/pull/263048)? Specifically, the disabling of `catchConflicts` for the `doCheck` passthru test.

Also is it specifically protobuf that causes conflicts?

**Review by samuela — APPROVED** [review]
I'd really prefer for us to debug and resolve the underlying duplicate packages issue. (I'm guessing it might have something to do with all the package overrides for tensorflow?)

But I think this PR is an improvement in the sense that it fixes the build and documents the problem in a code comment.

File: pkgs/development/python-modules/objax/default.nix (hunk 1 above)
Comment (Madouura) [reply pkgs/development/python-modules/objax/default.nix:45]: You can also just make passthru.tests.check and ``dontInstall``.
Since the purpose is for testing, and it's  a passthru attribute, you can work around the conflict without affecting the actual used package.

Comment (Madouura) [reply pkgs/development/python-modules/objax/default.nix:45]: Unfortunately, what I just suggested is not applicable here.
It was something that was only working because the conflict was direct and not indirect like this is.

Comment (GaetanLepage) [reply pkgs/development/python-modules/objax/default.nix]: In this case, there is no direct need for the `passthru` trick (no circular dependency problem). Hence, I have just set `catchConflicts` to `false` in the main scope. Is that a problem ?

**Comment by GaetanLepage** [comment]
I ended up re-enabling the tests, skipping the one needing internet and patching the broken ones.
Made a [PR upstream](https://github.com/google/objax/pull/266) about those broken ones.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 264463` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.objax</li>
    <li>python310Packages.objax.dist</li>
    <li>python311Packages.objax</li>
    <li>python311Packages.objax.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/objax/default.nix [hunk 2]
```diff
@@ -40,6 +51,23 @@ buildPythonPackage rec {
...
+
+  nativeCheckInputs = [
+    keras
+    pytestCheckHook
+    tensorflow
+  ];
+
+  pytestFlagsArray = [
+    "tests/*.py"
+  ];
```
Comment (samuela) [inline pkgs/development/python-modules/objax/default.nix:64]: why is this necessary? i would imagine it ought to default to something like this?

File: pkgs/development/python-modules/objax/default.nix [hunk 3]
```diff
@@ -40,6 +51,23 @@ buildPythonPackage rec {
     "objax"
   ];
 
+  catchConflicts = false;
```
Comment (samuela) [inline pkgs/development/python-modules/objax/default.nix:54]: let's add a code comment explaining this setting, since it's a little unconventional

File: pkgs/development/python-modules/objax/default.nix (hunk 2 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/objax/default.nix:64]: I thought too, but it is not the case (it fails to collect any test otherwise).
The script that they use to run the tests does the same thing: https://github.com/google/objax/blob/master/tests/run_tests.sh#L23

File: pkgs/development/python-modules/objax/default.nix (hunk 3 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/objax/default.nix:54]: Is that a problem to leave that in the main scope ? Or should we declare a `passthru.tests` attrs ?

Comment (GaetanLepage) [reply pkgs/development/python-modules/objax/default.nix:56]: I added a comment. Feel free to suggest another phrasing.

Comment (samuela) [reply pkgs/development/python-modules/objax/default.nix:56]: LGTM!

File: pkgs/development/python-modules/objax/default.nix (hunk 2 above)
Comment (samuela) [reply pkgs/development/python-modules/objax/default.nix:66]: Huh, weird

Comment (natsukium) [reply pkgs/development/python-modules/objax/default.nix:66]: The file names are so bad. This is the only way to do it.
https://docs.pytest.org/en/7.2.x/explanation/goodpractices.html#conventions-for-python-test-discovery

**Comment by natsukium** [comment]
ofborg says
```
trying https://github.com/google/objax/pull/266/commits/3058f699dc2cd5184a9488962fb3763f905d4236.patch
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 406
error: cannot download 3058f699dc2cd5184a9488962fb3763f905d4236.patch from any mirror
error: builder for '/nix/store/w8piy17dgxplgczz8lwi6f3ygj1fhlsy-3058f699dc2cd5184a9488962fb3763f905d4236.patch.drv' failed with exit code 1;
```

**Comment by GaetanLepage** [comment]
> ofborg says

Indeed, I forced pushed on my upstream commit.
Now fixed.

**Comment by GaetanLepage** [comment]
@ofborg build python39Packages.objax python310Packages.objax python311Packages.objax

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 264463` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.objax</li>
    <li>python310Packages.objax.dist</li>
    <li>python311Packages.objax</li>
    <li>python311Packages.objax.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
